validate.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -e
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# "---------------------------------------------------------"
+# "-                                                       -"
+# "-  Validation script checks if Cassandra                -"
+# "-  deployed successfully.                               -"
+# "-                                                       -"
+# "---------------------------------------------------------"
+
+# bash "strict-mode", fail immediately if there is a problem
+set -euo pipefail
+
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck disable=SC1090
+source "$ROOT"/common.sh
+
+APP_MESSAGE="successfully rolled out"
+
+# Get the kubectl credentials for the GKE cluster.
+gcloud container clusters get-credentials "$CLUSTER_NAME" --zone "$ZONE" --project "$PROJECT"
+
+# Wait for the rollout of hello-server to finish
+kubectl --namespace default rollout status --watch=false deployment/hello-server \
+ | grep "$APP_MESSAGE" &> /dev/null || exit 1
+echo "step 1 of the validation passed."
+
+# Grab the external IP and port of the service to confirm that hello-server deployed correctly.
+EXT_IP=""
+EXT_PORT=""
+while true
+do
+  sleep 1
+  EXT_IP="$(kubectl --namespace default get svc hello-server -ojsonpath='{.status.loadBalancer.ingress[0].ip}')"
+  EXT_PORT=$(kubectl --namespace default get service hello-server -o=jsonpath='{.spec.ports[0].port}')
+
+  if [[ $EXT_IP =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ && $EXT_PORT =~ ^[0-9]+$ ]]; then
+    break
+  else
+    continue
+  fi
```

Comment:
This is an infinite loop. Please test for like 10 min then break the loop.

Reply:
or set a retry count like in istio-shared/verify-functions.sh?